client/src/templates/Challenges/redux/current-challenge-saga.js

```diff
@@ -33,13 +36,27 @@ export function* currentChallengeSaga({ payload: id }) {
   }
 }
 
-function* updateSuccessMessageSaga() {
+function* showDonateModalSaga() {
+  /* wait for the newchallenge to be mounted to avoid mounting
+    the donation modal twice, once for the old chall and once for the new one
+    */
+  let shouldRequestDonation = yield select(shouldRequestDonationSelector);
+  if (shouldRequestDonation) {
+    yield take(challTypes.closeModal);
+    yield delay(1000);
```

Comment:
I think this could break if the user has a slow connection and takes longer than a second to mount the next page.

I'm not sure what the perfect solution is, though. I just feel like the donation modal should be triggered by the mounting of a new page, rather than the closing of another modal.

Reply:
Yes, you are right. previously it used to wait for challengeMounted action, but that cannot be used now since the next page after the last challenge in a block is an into and challangeMounted will not be called. Maybe we should call a introMounted action for the intros and listen to that for this donation modal. what do you think?

Reply:
note to self, there are times when the user has finished the last challenges but the last remaining challenge is in the middle of the block meaning closing the completion modal will be followed by challenge Mounted.

Reply:
How about putting it in the Learn layout?  That way, no matter where the user ends up after completing a challenge, as long an they're still in /learn they'll see the modal.  I think that might be more robust than checking in challenges and intros.